dashboard/app/models/programming_method.rb

```diff
@@ -20,4 +20,28 @@
 #  index_programming_methods_on_key_and_programming_class_id  (key,programming_class_id) UNIQUE
 #
 class ProgrammingMethod < ApplicationRecord
+  include CurriculumHelper
+
+  belongs_to :programming_class
+
+  before_validation :generate_key, on: :create
+  validates_uniqueness_of :key, scope: :programming_class_id, case_sensitive: false
+  validate :validate_key_format
+
+  def generate_key
+    return key if key
+    key = ProgrammingMethod.sanitize_key(name)
+    self.key = key
+  end
+
+  def self.sanitize_key(str)
```

Comment:
Could you add a comment describing what this should do 😬  -- kind of covered in test (and maybe obvious to ep tools folks), but would be nice to amateurs like myself

Reply:
Yes definitely! I'm doing some weird stuff relative to other models here as well so I definitely should've added a comment here...